x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,402 @@
+              'Elastic Crawler must be removed. For full details, see the documentation. ',
+          }),
+        },
+        documentationUrl: docsUrl,
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.enterpriseSearch.deprecations.crawler.listConnectors', {
+              defaultMessage: 'Enumerate all connector records',
+            }),
+            i18n.translate('xpack.enterpriseSearch.deprecations.crawler.deleteCrawlers', {
+              defaultMessage: 'Delete any that have `service_type: elastic-crawler`',
+            }),
+          ],
+          api: {
+            method: 'POST',
+            path: '/internal/enterprise_search/deprecations/delete_crawler_connectors',
+            body: {
+              ids: crawlers.map((it) => it.id),
+            },
+          },
+        },
+      },
+    ];
+  }
+}
+
+/**
+ * if the customer is using Native Connectors, agentless is available, but the integration server is missing, they are told that Integrations Server must be added
+ * if the customer is using Native Connectors, and agentless is unavailable, they are told that they must convert their connectors to Connector Clients
+ * if the customer was using "native" connectors that don't match our connector service types, they must delete them or convert them to connector clients.
+ */
+export async function getNativeConnectorDeprecations(
+  ctx: GetDeprecationsContext,
+  hasAgentless: boolean,
+  hasFleetServer: boolean,
+  cloud: CloudSetup,
+  docsUrl: string
+): Promise<DeprecationsDetails[]> {
+  const client = ctx.esClient.asInternalUser;
+  const connectors: Connector[] = await fetchConnectors(client, undefined, false, undefined);
+  const nativeConnectors = connectors.filter((hit) => hit.is_native);
+  if (nativeConnectors.length === 0) {
+    return []; // no deprecations to register if there are no Native Connectors
+  } else {
+    const deprecations: DeprecationsDetails[] = [];
+
+    if (nativeConnectors.length > 0 && !hasAgentless) {
+      // you can't have elastic-managed connectors in an environment that Elastic doesn't manage
+      // ... and agentless is available in every Elastic-managed environment
+
+      deprecations.push({
+        level: 'critical',
+        deprecationType: 'feature',
+        title: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.title', {
+          defaultMessage:
+            'Connectors with `is_native: true` are not supported in self-managed environments',
+        }),
+        message: {
+          type: 'markdown',
+          content: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.message', {
+            defaultMessage:
+              '"Native Connectors" are managed services in Elastic-managed environments such as Elastic Cloud Hosted and ' +
+              'Elastic Serverless. Any connectors with `is_native: true` must be converted to connector clients or deleted ' +
```

Comment:
```suggestion
              'Elastic Cloud Serverless. Any connectors with `is_native: true` must be converted to connector clients or deleted ' +
```

hard to keep up with official naming policies :)